libs/stats/odc/stats/_pq.py

```diff
@@ -21,14 +21,9 @@
     "thin cirrus",
 )
 
-# Filters are a list of dict(closing=int, opening=int, dilation=int), where
-# closing(Optional) = remove small holes in cloud - morphological closing
-# opening = shrinks away small areas of the mask
-# dilation = adds padding to the mask
-#
-# For each entry in the list an extra ``.clear_{r1}_{r2}`` band will be added to the output in addition to
-# ``.total`` and ``.clear`` bands that are always computed.
-default_filters = [dict(opening=2,dilation=5), dict(opening=0,dilation=5)]
+# filters = list of iterable tuples of morphological operations in the order you want them performed,
+# e.g. [[("opening", 2), ("dilation", 5)]]
+default_filters = [[("opening", 2), ("dilation", 5)], [("opening", 0), ("dilation", 5)]]
```

Comment:
This is now confusing, it used to be a list of tuples, now it's a list of a list of tuples. Each element of an outer list is a sequence of operations to apply to mask.

It's probably best to change this to a dictionary, `Dict[str, Iterable[Tuple[str, int]]` and then use the key to define output band name for a given configuration.

Reply:
> This is now confusing, it used to be a list of tuples, now it's a list of a list of tuples. Each element of an outer list is a sequence of operations to apply to mask.
> 
> It's probably best to change this to a dictionary, `Dict[str, Iterable[Tuple[str, int]]` and then use the key to define output band name for a given configuration.

@Kirill888 Is this easy to read?

```
default_filters = [{
    "clear_2_5": [("opening", 2), ("dilation", 5)], 
    "clear_0_5": [("opening", 0), ("dilation", 5)]
}]
```

Reply:
@NikitaGandhi I think so, also removes the need for building up those output band names in code. Not 100% sure if it is equivalent to what it was before the change. Was it `opening|>dilation`, or was it `opening|>closing`? Also not sure how much this matters.